hca/dss/__init__.py

```diff
@@ -361,20 +372,25 @@ def _download_manifest_bundle(self,
             logger.info('Primary copies of the files have been downloaded to `.hca` and linked '
                         'into per-bundle subdirectories of the current directory.')
 
-    def _download_tasks(self,
-                        bundle_uuid,
-                        replica,
-                        version="",
-                        download_dir="",
-                        metadata_files=('*',),
-                        data_files=('*',),
-                        num_retries=10,
-                        min_delay_seconds=0.25):
-        bundle = self.get_bundle(uuid=bundle_uuid, replica=replica, version=version if version else None)["bundle"]
-        bundle_fqid = bundle_uuid + '.' + bundle['version']
+    def _bundle_download_tasks(self,
+                               bundle_uuid,
+                               replica,
+                               version="",
+                               download_dir="",
+                               metadata_files=('*',),
+                               data_files=('*',),
+                               num_retries=10,
+                               min_delay_seconds=0.25):
+        """
+        Returns an iterator of tasks that each download one of the files in a bundle.
+        """
+        logger.info('Downloading bundle %s version %s ...', bundle_uuid, version)
+        bundle = self.get_bundle.iterate(uuid=bundle_uuid, replica=replica, version=version if version else None)
```

Comment:
Switch to `.paginate()` which was added yesterday. Sorry for not alerting you earlier.